### PR TITLE
Allowed ContentletRelationships type to be copied for contentlet

### LINKED ISSUE #24167: Relationship fields couldn't update with workflow apis
### Problem Statement

When workflow action apis use to update/add relationship fields for a contentlet then there are two behaviors occur.

1. It returns a message with Exception If space uses between queries.
Example:  
```
"contentlet": {
    "regions":"(+contentType:Region) +identifier:994d34aaf93f98c90a666d6fef537812"
  }
  
  OR
  
   "contentlet": {
    "regions":"(+contentType:Region) +Region.title:('PSGNU058')"
  }
```

Response:

`{"message":"The map contains an invalid value: class com.dotmarketing.portlets.structure.model.ContentletRelationships"}`

2: It returns contentlet data but didn't update relationship field however the last update date change for contenlet

Example:
```

 "contentlet": {
  "regions":"(+contentType:Region)+Region.title:*Asia*"
  }


```
Response:
Contenlet data in json -- Omit that response for bravery 

### Steps to Reproduce

Create two contenttypes e.g Movie, Region

Create a relationship field one-to-many in Movie with Region

Create two Region contentlets e.g Asia, Africa

Create a contenlet of Movie and don't assign region to it yet because try to add with Workflow API

At last do REST api call with correct identifier and inode

```
curl -X 'PUT' \
  'https://demo.dotcms.com/api/v1/workflow/actions/b9d89c80-3d88-4311-8365-187323c96436/fire?identifier=33c53dd87135947b76147a347a765424&language=-1' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "contentlet": {
    "regions":"+contentType:Region +Region.title:*Asia*"
  }
}'
```

### Acceptance Criteria

If REST body is correct then Workflow REST api should update relationship field according to data otherwise should return meaning full error response.

### dotCMS Version

23.0.1

### Proposed Objective

Customer Support

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/110108

### Assumptions & Initiation Needs

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* Fix to allow workflow action API to create relationship

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **
image

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

**Review by nollymar — CHANGES_REQUESTED** [review]
Ideally, this change should have a test case in the ESContentletAPIImplTest and the postman test implemented should describe the real case fixed, especially because this is an issue caught by a customer.

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/WorkflowAPITest.java [hunk 1]
```diff
@@ -4070,4 +4069,91 @@ public void saveWorkflowTask() throws DotDataException {
 
     }
 
+
```
Comment (erickgonzalez) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/WorkflowAPITest.java:4072]: If this is intended to be commented, let's removed

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

**Comment by jdotcms** [comment]
> Ideally, this change should have a test case in the ESContentletAPIImplTest and the postman test implemented should describe the real case fixed, especially because this is an issue caught by a customer.

Hi @nollymar not sure what you mean about real case, Hassan has created a post man test that does exactly that the issue described:
1) create a couple content types
2) Related between them
3) Create the instances
4) ask for the relationship

Regarding the Integration test @nollymar and @erickgonzalez, I am not sure why it is commented but I will take a look

**Review by nollymar — APPROVED** [review]

**Comment by fmontes** [comment]
we need to fix this and merge with master

**Pull request merged**